src/Simulation/Simulators/NewTracer/MetricCollectors/DepthCounterBase.cs

```diff
@@ -111,30 +145,33 @@ void IQubitTrackingTarget.OnReleaseQubits(IQArray<Qubit> qubits)
             this.CurrentState.ReleasedQubitsTime = System.Math.Max(this.CurrentState.ReleasedQubitsTime, releasedTime);
         }
 
-        //TODO: implement time storing as dictionary id->time an clear entries on release
-
         void IQubitTrackingTarget.OnAllocateQubits(IQArray<Qubit> qubits)
         {
+            foreach (Qubit qubit in qubits) {
+                this.MaxQubitId = System.Math.Max(this.MaxQubitId, qubit.Id);
+            }
```

Comment:
```suggestion
            this.MaxQubitId = System.Math.Max(this.MaxQubitId, qubits.Max(qubit => qubit.Id));
```

Reply:
Updated formatting.
Loop makes it clear each qubit is touched. Lambda function adds unnecessary complexity to the code and hides computation complexity of the algorithm.